s3tests_boto3/functional/test_s3.py

```diff
@@ -764,6 +1215,16 @@ def test_bucket_list_marker_not_in_list():
     keys = _get_keys(response)
     eq(keys, ['foo', 'quxx'])
 
+def test_bucket_list_startafter_not_in_list():
+    key_names = ['bar', 'baz', 'foo', 'quxx']
+    bucket_name = _create_objects(keys=key_names)
+    client = get_client()
+
+    response = client.list_objects_v2(Bucket=bucket_name, StartAfter='blah')
+    eq(response['StartAfter'], 'blah')
+    keys = _get_keys(response)
+    eq(keys, ['foo', 'quxx'])
```

Comment:
why are the keys here only `['foo', 'quxx']` and not the whole list in `key_names`.

Reply:
These are the keynames: ['bar', 'baz', 'foo', 'quxx'] and we are specifying startafter as blah. Amazon S3 starts listing after this specified key.  'bar' and 'baz' starts with 'b' of 'blah', but doesn't match with 'blah', It starts looking from 'foo'. Here even the first letter is not matching. Hence the keys remaining are 'foo' and 'quxx'. We could verify this by trying for different values for StartAfter.